R/plot_fns.R

```diff
@@ -4128,304 +4164,251 @@ plot.rmdFilt <- function(x, pvalue_threshold = NULL, sampleID = NULL,
+        alpha = .8,
+        shape = 4,
+        color = "black",
+        size = point_size + 1
+      ) + ggplot2::labs(fill = "")
+    }
+    
+    
+    ## Add hline ##
+    if(!is.null(pvalue_threshold)){
+      p <- p + ggplot2::geom_hline(yintercept = yint)
+    }
+    
+    ## Remove unneeded legends ##
+    
+    ## Shape ##
+    if (all(filter_object$pch == "")) {
+      p <- p +
+        ggplot2::guides(shape = "none")
+    }
+    
+    
+    # Farm boy, remove the default legend(s) if the data are paired and there are
+    # no main effects. As you wish.
+    # If there are no main effects do not print a legend.
+    if ("paired_diff" %in% attr(filter_object, "group_DF")$Group && 
+        length(color_by) == 1 &&
+        color_by == "Group") {
+      # Remove the color legend. This legend is automatically created because we
+      # specified a variable to color by in aes().
+      p <- p +
+        ggplot2::guides(color = "none")
+    }
 
-    # Start scatter plot skeleton when a p-value threshold is specified.
-    p <- ggplot2::ggplot(filter_object)
-
-    # Start scatter plot skeleton when a p-value threshold has been provided.
-    if (length(main_eff_names) == 1) {
-      # we need separate plots for if we have pair id or not
-
-      if (!is.null(attr(attr(filter_object, "group_DF"), "pair_id"))) {
-        # if !null then we need to include pair fill information
-        p <- p +
-          ggplot2::geom_point(
-            ggplot2::aes(
-              x = !!dplyr::sym(samp_id),
-              y = Log2.md,
-              col = !!dplyr::sym(main_eff_names),
-              # Add a fill layer that will not affect how the plot looks. This
-              # layer is used to create a legend when one sample from a pair is
-              # an outlier but the other sample belonging to the pair is not.
-              fill = "Removed because paired with outlier"
-            ),
-            alpha = ifelse(goodies_alpha, 1, 0.5),
-            size = point_size,
-            shape = ifelse(goodies_pch, 15, 16)
-          )
-      } else {
-        # if null pair id then we do not need to account for fill
-        p <- p +
-          ggplot2::geom_point(
-            ggplot2::aes(
-              x = !!dplyr::sym(samp_id),
-              y = Log2.md,
-              col = !!dplyr::sym(main_eff_names)
-            ),
-            alpha = ifelse(goodies_alpha, 1, 0.5),
-            size = point_size,
-            shape = ifelse(goodies_pch, 15, 16)
-          )
-      }
+  } else {
+    
+    edit_label <- dfmelt$variable == "Proportion_Missing"
+    dfmelt$variable[edit_label] <- "Prop. Missing"
+    
+    ## Switch Proportion missing to prop missing
+    
+    ## Hist plot ##
+    if(length(color_by) != 1){
+
+      colorlabel <- toString(as.character(display_names))
+      
+      p <- dfmelt %>%
+        ggplot2::ggplot(ggplot2::aes(x = value, 
+                                     fill = paste(!!rlang::sym(color_by[1]),
```

Comment:
we do not import `rlang` to keep dependencies slim.  I think replacing with `dplyr::sym` should do the trick to remove the warnings.